tests/functional/watcher_controller_test.go

```diff
@@ -1435,4 +1482,219 @@ var _ = Describe("Watcher controller", func() {
+			DeferCleanup(th.DeleteInstance, CreateWatcher(watcherTest.Instance, spec))
+			DeferCleanup(k8sClient.Delete, ctx, CreateWatcherMessageBusSecret(watcherTest.Instance.Namespace, "rabbitmq-secret"))
+			memcachedSpec := memcachedv1.MemcachedSpec{
+				MemcachedSpecCore: memcachedv1.MemcachedSpecCore{
+					Replicas: ptr.To(int32(1)),
+				},
+			}
+			DeferCleanup(infra.DeleteMemcached, infra.CreateMemcached(watcherTest.Watcher.Namespace, MemcachedInstance, memcachedSpec))
+			infra.SimulateMemcachedReady(watcherTest.MemcachedNamespace)
+			DeferCleanup(
+				mariadb.DeleteDBService,
+				mariadb.CreateDBService(
+					watcherTest.Instance.Namespace,
+					*GetWatcher(watcherTest.Instance).Spec.DatabaseInstance,
+					corev1.ServiceSpec{
+						Ports: []corev1.ServicePort{{Port: 3306}},
+					},
+				),
+			)
+			DeferCleanup(keystone.DeleteKeystoneAPI, keystone.CreateKeystoneAPI(watcherTest.WatcherAPI.Namespace))
+			DeferCleanup(
+				k8sClient.Delete, ctx, th.CreateSecret(
+					types.NamespacedName{Namespace: watcherTest.Instance.Namespace, Name: "combined-ca-bundle"},
+					map[string][]byte{
+						"internal-ca-bundle.pem": []byte("some-b64-text"),
+						"tls-ca-bundle.pem":      []byte("other-b64-text"),
+					},
+				))
+			DeferCleanup(
+				k8sClient.Delete, ctx, th.CreateSecret(
+					types.NamespacedName{Namespace: watcherTest.Instance.Namespace, Name: "custom-prometheus-config"},
+					map[string][]byte{
+						"host":      []byte("customprometheus.example.com"),
+						"port":      []byte("9092"),
+						"ca_secret": []byte("combined-ca-bundle"),
+						"ca_key":    []byte("internal-ca-bundle.pem"),
+					},
+				))
+
+			mariadb.SimulateMariaDBAccountCompleted(watcherTest.WatcherDatabaseAccount)
+			mariadb.SimulateMariaDBDatabaseCompleted(watcherTest.WatcherDatabaseName)
+			infra.SimulateTransportURLReady(watcherTest.WatcherTransportURL)
+			DeferCleanup(
+				k8sClient.Delete, ctx, th.CreateSecret(
+					types.NamespacedName{Namespace: watcherTest.Instance.Namespace, Name: SecretName},
+					map[string][]byte{
+						"WatcherPassword": []byte("password"),
+					},
+				))
+
+			// simulate that it becomes ready i.e. the keystone-operator
+			// did its job and registered the watcher service
+			keystone.SimulateKeystoneServiceReady(watcherTest.KeystoneServiceName)
+
+			// Simulate dbsync success
+			th.SimulateJobSuccess(watcherTest.WatcherDBSync)
+
+			// Simulate WatcherAPI deployment
+			th.SimulateStatefulSetReplicaReady(watcherTest.WatcherAPIStatefulSet)
+
+			// Simulate KeystoneEndpoint success
+			keystone.SimulateKeystoneEndpointReady(watcherTest.WatcherKeystoneEndpointName)
+
+			// Simulate WatcherApplier deployment
+			th.SimulateStatefulSetReplicaReady(watcherTest.WatcherApplierStatefulSet)
+
+			// Simulate WatcherDecisionEngine deployment
+			th.SimulateStatefulSetReplicaReady(watcherTest.WatcherDecisionEngineStatefulSet)
+
+		})
+
+		//It("should have the Spec fields with the expected values", func() {
```

Comment:
why is this test commented out? It looks relevant to the PR

Reply:
Uncommented and fixed.